picard/file.py

```diff
@@ -247,8 +253,9 @@ def _saving_finished(self, result=None, error=None):
             self.clear_pending()
             self._add_path_to_metadata(self.orig_metadata)
 
-        del self.tagger.files[old_filename]
-        self.tagger.files[new_filename] = self
+        if self.state != File.REMOVED:
```

Comment:
Yes - we get here if file was removed **during** the save - and we only add the file to the tagger if it wasn't removed.